packages/api/test/pin.spec.js

```diff
@@ -676,6 +676,44 @@ describe('Pinning APIs endpoints', () => {
       assert.strictEqual(data.pin.cid, sourceCid)
       assert.notDeepEqual(data.status, 'failed')
     })
+
+    it('should throw if account is PSA enabled, but restricted', async () => {
```

Comment:
nit.

You've rightly added tests in the context of every single endpoint that has to be restricted.

I wonder if having a whole describe block for pinning authorization where we add these tests, would make them a bit more focused and allow you to avoid some code duplication.  

ie. you have the list of restrict-able tests and loop through them.
Not a big deal, just an idea.

Reply:
I actually debated whether we should have an `auth.spec.js` where we test all the auth things. Having all in one file might help with documentation too since all the expected behaviour will be there.What do you think?

Reply:
I like it!